pvlib/iotools/cams.py

```diff
@@ -0,0 +1,357 @@
+
+    The returned data DataFrame includes the following fields:
+
+    =======================  ======  ==========================================
+    Key, mapped key          Format  Description
+    =======================  ======  ==========================================
+    **Mapped field names are returned when the map_variables argument is True**
+    ---------------------------------------------------------------------------
+    Observation period       str     Beginning/end of time period
+    TOA, ghi_extra           float   Horizontal radiation at top of atmosphere
+    Clear sky GHI, ghi_clear float   Clear sky global radiation on horizontal
+    Clear sky BHI, bhi       float   Clear sky beam radiation on horizontal
+    Clear sky DHI, dhi_clear float   Clear sky diffuse radiation on horizontal
+    Clear sky BNI, dni_clear float   Clear sky beam radiation normal to sun
+    GHI, ghi*                float   Global horizontal radiation
+    BHI, bhi*                float   Beam (direct) radiation on horizontal
+    DHI, dhi*                float   Diffuse horizontal radiation
+    BNI, dni*                float   Beam (direct) radiation normal to the sun
+    Reliability*             float   Fraction of reliable data in summarization
+    =======================  ======  ==========================================
+
+    *Parameters only returned if service='cams_radiation'. For description of
+    additional output parameters in verbose mode, see [1]_ and [2]_.
+
+    Note that it is recommended to specify the latitude and longitude to at
+    least the fourth decimal place.
+
+    Variables corresponding to standard pvlib variables are renamed,
+    e.g. `sza` becomes `solar_zenith`. See the
+    `pvlib.iotools.cams.CAMS_RADIATION_VARIABLE_MAP` dict for the complete
+    mapping.
+
+    See Also
+    --------
+    pvlib.iotools.read_cams_radiation, pvlib.iotools.parse_cams_radiation
+
+    Raises
+    ------
+    requests.HTTPError
+        If the request is invalid, then an XML file is returned by the CAMS
+        service and the error message will be raised as an expcetion.
+
+    References
+    ----------
+    .. [1] `CAMS Radiation Service Info
+       <http://www.soda-pro.com/web-services/radiation/cams-radiation-service/info>`_
+    .. [2] `CAMS McClear Service Info
+       <http://www.soda-pro.com/web-services/radiation/cams-mcclear/info>`_
+    .. [3] `CAMS McClear Automatic Access
+       <http://www.soda-pro.com/help/cams-services/cams-mcclear-service/automatic-access>`_
+    """
+    if time_step in TIME_STEPS_MAP.keys():
+        time_step_str = TIME_STEPS_MAP[time_step]
+    else:
+        warnings.warn('Time step not recognized, 1 hour time step used!')
+        time_step, time_step_str = '1h', 'PT01H'
+
+    if (verbose is True) & ((time_step != '1min') | (time_ref != 'UT')):
+        verbose = False
+        warnings.warn("Verbose mode only supports 1 min. UT time series!")
```

Comment:
I'd probably also raise a ValueError here, but that's debatable.

Reply:
I would prefer to only raise a warning, so if this is ok, I will leave it as it is.

Reply:
so users will see this warning and then hit `raise requests.HTTPError(errors, response=res)` on line 210?